code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+		if(mindTypeTwo == "Higher" || mindTypeTwo == "Artificial")
+			var/list/woopsTwo = list()
+			M = occupantTwo
+			for(var/mob/living/R in mob_list)
+				if((!R.mind) && (R.stat != 2) && (get_dist(src, R) <50) && (connectTwo.z == R.z) && (!is_type_in_list(R, illegalSwap)))
+					woopsTwo += R
+			if(woopsTwo.len)
+				var/list/M_spells = list()
+				M_spells = M.spell_list.Copy()
+				woopsTarget = pick(woopsTwo)
+				if(M.mind.special_verbs.len)
+					for(var/V in M.mind.special_verbs)
+						M.verbs -= V
+				for(var/spell/S in M.spell_list)
+					M.remove_spell(S)
+				M.mind.transfer_to(woopsTarget)
+				for(var/spell/S in M_spells)
+					woopsTarget.add_spell(S)
+				to_chat(M, "<span class='danger'>Your mind is thrown out of the machine and forced into a nearby vessel</span>")
+				playsound(M, "sound/effects/phasein.ogg", 50, 1)
+				errorMessage = "Error in mind delivery"
+			else
+				to_chat(M, "<span class='danger'>Your mind is severely damaged by the feedback</span>")
+				playsound(M, "sound/misc/balloon_pop.ogg", 50, 1)
+				if(ishuman(M))
+					M.adjustBrainLoss(75)
+				errorMessage = "Mind sustained damage"
+		if(mindTypeOne == "Lower")
+			M = occupantOne
+			M.faction = pick(randFaction)
+		if(mindTypeTwo == "Lower")
+			M = occupantTwo
+			M.faction = pick(randFaction)
+	else
+		soulShardSafety = FALSE
+		errorMessage = "Safety consumed"
+	currentlySwapping = FALSE
+	occupantScan = FALSE
+	connectOne.icon_state = "mind_pod_closed"
+	connectTwo.icon_state = "mind_pod_closed"
+	icon_state = "mind_hub"
+	swapProgress = 0
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/theFlySwap(var/mob/living/MO, var/mob/living/MT)
+	var/list/mindFly = list()
+	var/list/simpFly = list()
+	MO = occupantOne
+	MT = occupantTwo
+	for(var/obj/item/weapon/holder/D in theFly)
+		for(var/mob/M in D.contents)
+			theFly += M //Gets the mob stored in the holder, adds it to the list
+		D.Destroy() //Turns them from holder to mob
+	theFly += MO
+	theFly += MT
+	for(var/mob/living/q in theFly)
+		if(q.mind)
+			mindFly += q
+			theFly -= q
+		else
+			simpFly += q
+			theFly -=q
+	var/mob/living/dummy = new(src.loc)
+	simpFly += dummy
+	var/mob/nuYou = null
+	for(var/mob/living/flies in mindFly)
+		nuYou = pick(simpFly)
+		if(flies.mind.special_verbs.len)
+			for(var/V in flies.mind.special_verbs)
+				flies.verbs -= V
+		for(var/spell/S in flies.spell_list)
+			flies.remove_spell(S)
+		flies.mind.transfer_to(nuYou)
+		simpFly += flies
+		simpFly -= nuYou
+	if(dummy.mind)
+		nuYou = pick(simpFly)
+		dummy.mind.transfer_to(nuYou)
+	qdel(dummy)
+	currentlySwapping = FALSE
+	occupantScan = FALSE
+	theFly.len = 0
+	simpFly.len = 0
+	mindFly.len = 0
+	connectOne.icon_state = "mind_pod_closed"
+	connectTwo.icon_state = "mind_pod_closed"
+	icon_state = "mind_hub"
+	badSwap = 1
+	swapProgress = 0
+	errorMessage = "Mind transport anomaly"
+
+//Swaps over///////////////
+//Pod stuff//////////
+
+/obj/machinery/mind_machine/mind_machine_pod
```

Comment:
You should hook `Exited` on these pods to make sure the contained mob gets updated in case the player leaves the pod *somehow* (I think there are items that can teleport you out like this even when it's locked?)